src/ui/components/modals/SurveyModal.native.test.tsx

```diff
@@ -11,28 +11,42 @@ const hideModalMock = jest.fn()
 
 describe('<SurveyModal />', () => {
   it('should redirect to survey when pressing "Répondre au questionnaire" button', () => {
-    renderSurveyModal()
+    renderSurveyModal({ surveyUrl: 'https://fr.wikipedia.org/wiki/FIEALD' })
 
     fireEvent.press(screen.getByText('Répondre au questionnaire'))
 
     expect(openUrl).toHaveBeenCalledWith('https://fr.wikipedia.org/wiki/FIEALD', undefined, true)
   })
 
   it('should call hideModal function when pressing close icon', () => {
-    renderSurveyModal()
+    renderSurveyModal({})
     const rightIcon = screen.getByTestId('Fermer la modale')
     fireEvent.press(rightIcon)
 
     expect(hideModalMock).toHaveBeenCalledTimes(1)
   })
+
+  it('should display "Répondre au questionnaire" button when survey url defined', () => {
+    renderSurveyModal({ surveyUrl: 'https://fr.wikipedia.org/wiki/FIEALD' })
+
+    expect(screen.getByText('Répondre au questionnaire')).toBeOnTheScreen()
+  })
+
+  it('should not display "Répondre au questionnaire" button when survey url not defined', () => {
+    renderSurveyModal({})
+
+    expect(screen.queryByText('Répondre au questionnaire')).not.toBeOnTheScreen()
+  })
 })
 
-const renderSurveyModal = () => {
+type RenderSurveyModalType = Partial<ComponentProps<typeof SurveyModal>>
```

Comment:
```suggestion
type RenderSurveyModalType = Pick<ComponentProps<typeof SurveyModal>, 'surveyUrl'>
```
Techniquement, c'est plutôt ça le typage non ?

Reply:
Tu as raison, soyons précis :)